coderd/database/dbrollup/dbrollup.go

```diff
@@ -0,0 +1,127 @@
+package dbrollup
```

Comment:
I see a potential abstraction here -- the PR I'm working on in parallel https://github.com/coder/coder/pull/12659 has a similar logic of "run a query every interval".  There are definitely other areas of the codebase that could benefit from a similar framework.